lib/OpenQA/Script/CloneJob.pm

```diff
@@ -105,7 +105,8 @@ sub clone_job_get_job {
 
 sub clone_job_download_assets {
     my ($jobid, $job, $remote, $remote_url, $ua, $options) = @_;
-    my @parents = map { clone_job_get_job($_, $remote, $remote_url, $options) } @{$job->{parents}->{Chained}};
+    my $job_type = $options->{'add-childrens'} ? 'children' : 'parents';
```

Comment:
"children" is already plural (of "child"). So please remove the extra "s". And maybe `--clone-children` or `--include-children` would be better.

I'm also wondering why the flag changes this part of the code. Maybe that's something which is useful but alone from the flags name "add-children" is is very unclear and unintuitive why the dependency traversal order would be reversed here.

Reply:
> I'm also wondering why the flag changes this part of the code. Maybe that's something which is useful but alone from the flags name "add-children" is is very unclear and unintuitive why the dependency traversal order would be reversed here.

What do you mean? The main part of the change is this reverse, as currently if you try to clone a node only the supportserver (the parent job) is cloned, not the other nodes, because these jobs are children of the supportserver. So the idea is to clone the supportserver with this new option to ensure that all dependencies are cloned.
And later in the code we can see another big change: children jobs should be cloned *after* the supportserver, to be sure to have the correct dependencies.

Reply:
> What do you mean?

The function you're modifying here is about the asset download of chained dependencies. I'm just not sure how that relates to cloning parallel siblings as well.

> if you try to clone a node only the supportserver (the parent job) is cloned, not the other nodes, because these jobs are children of the supportserver.

I know. The clone script so far only "goes up" the dependency chain. I'm just wondering why this function which is about the asset download would need to be reversed for that.

And in the code down you're also considering children *in addition* to parents. That makes sense because this part of the behavior shouldn't change. However, you're change in this function does *not* consider children in addition so parents are not traversed anymore like before. Maybe that even makes sense as well but then an explaining comment in the code would be helpful.

Reply:
> The function you're modifying here is about the asset download of chained dependencies. I'm just not sure how that relates to cloning parallel siblings as well.

Sorry I thought it was in the other function. So yes you're right I have to double check, maybe this is not needed in fact. I will do tests for this.

Reply:
Yes, it seems that this change is not needed, so I revert it.